XrmToolBox.Controls/Controls/CRMGridView.cs

```diff
@@ -672,7 +672,7 @@ private void AddColumnForAttribute(IEnumerable<Entity> entities, List<DataColumn
             if (CreateColumnForAttribute(entities, attribute, force) is DataColumn dataColumn && dataColumn != null)
             {
                 var meta = dataColumn.ExtendedProperties.ContainsKey("Metadata") ? dataColumn.ExtendedProperties["Metadata"] as AttributeMetadata : null;
-                if (!force && meta?.IsPrimaryId == true)
+                if (meta?.IsPrimaryId == true && (!force || ShowIdColumn && meta.LogicalName == attribute))
```

Comment:
Excellent if the comment is updated to explain the quite intricate boolean excursions here :) @MarkMpn

Reply:
Updated in https://github.com/jamesnovak/xrmtb.XrmToolBox.Controls/pull/13/commits/8da4985bfd0e534362f056f6618a133c1153ddb6